service/tbc/tbc.go

```diff
@@ -2656,10 +2683,37 @@ func (s *Server) synced(ctx context.Context) (si SyncInfo) {
 			Height: keystoneBH.Height,
 		}
 		si.Keystone = *keystoneHH
-		if keystoneHH.Hash.IsEqual(&bhb.Hash) {
+	}
+
+	if s.cfg.ZKIndex {
+		// Perform additional zk indexer tests.
+		zkBH, err := s.zki.IndexerAt(ctx)
+		if err != nil {
+			zkBH = &tbcd.BlockHeader{}
+		}
+		zkHH := &HashHeight{
+			Hash:   zkBH.Hash,
+			Height: zkBH.Height,
+		}
+		si.ZK = *zkHH
```

Comment:
Is there any reason to create a `zkHH` variable pointer here and then dereference it?
```suggestion
		si.ZK = HashHeight{
			Hash:   zkBH.Hash,
			Height: zkBH.Height,
		}
```

Reply:
Yes, it uses the return value of  `zkBH, err := s.zki.IndexerAt(ctx)` to create the variable. We just reuse it later if an error occurred. It's dereference there or elsewhere.